src/test/java/BaseTest.java

```diff
@@ -1,11 +1,260 @@
+        playListFromSideBar.click();
+    }
+
+    protected void doubleClickAPlaylist() {
+        action = new Actions(driver);
+        WebElement playListFromSideBar = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector(".playlist:nth-child(5)")));
+        action.doubleClick(playListFromSideBar);
+    }
+
+
+    protected void deleteSelectedPlayList() {
+        WebElement deleteButton = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector(".btn-delete-playlist")));
+        deleteButton.click();
+    }
+
+    protected void managePopUpWindow() {
+        //WebElement confirm = driver.findElement(By.cssSelector("button.ok"));
+        //f (Assert.assertTrue(driver.findElement(By.cssSelector("button.ok")).isDisplayed()) {
+        //driver.findElement(By.cssSelector("button.ok")).click();
+        // confirm.click();
+        //}
+        try {
+            WebElement confirm = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("button.ok")));
+            confirm.click();
+        } catch(Exception e) {
+
+        }
+    }
+    protected String getSongDeletedFromPlaylistNotification() {
+        WebElement deletedFromPlaylistNotification = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("div.success.show")));
+        return deletedFromPlaylistNotification.getText();
+    }
+
+    //count songs helper methods
+    protected void choosePlaylistByName(String playlistName) {
+    wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//a[contains(text(),'" + playlistName + "')]"))).click();
+    }
+
+    public int countSongs() {
+    return driver.findElements(By.cssSelector("section#playlistWrapper td.title")).size() ;
+    }
+
+    protected String getPlaylistDetails() {
+    return wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("span.meta.text-secondary span.meta"))).getText();
+    }
+
+    protected void displayAllSongs() {
+        List<WebElement> songList = driver.findElements(By.cssSelector("section#playlistWrapper td.title"));
+        System.out.println("Number of Songs Found:" + countSongs());
+        for (WebElement e: songList) {
+            System.out.println(e.getText());
+        }
+    }
+
+    //rename playlist helper methods
+    String newPlayListName = "re-named";
+    protected void enterNewPlayListName(String newPlayListName) {
+        WebElement playlistInputField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[name='name']]")));
+        playlistInputField.sendKeys(Keys.chord(Keys.CONTROL, "A", Keys.BACK_SPACE));
+        playlistInputField.sendKeys(newPlayListName);
+        playlistInputField.sendKeys(Keys.ENTER);
+    }
+
+    public boolean doesPlayListExist() {
+        WebElement playListElement = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//a[text()='" + newPlayListName + "']")));
+        return playListElement.isDisplayed();
+    }
+
+
+
+
+
+
```

Comment:
**Please delete the empty line!** An Empty Line is a "Code Smell"
The short answer is, _They’re more an annoyance than a problem.They won’t actually break anything—which is why your code continues to run—but they make it awkward to read and change your code, especially if you’re dealing with multiple line breaks per line of real code. They also add just a tiny bit of size to your files, which is nice to reduce._